### PR TITLE
Move engine to shade package and update docs.

### PR DESCRIPTION
**Pull request by hurricanerix**

### CODE REVIEW

File: engine.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+// Engine contains state for the core systems of a 2.5D game.
+type Engine struct {
+	Title string
+}
+
+// New returns a pointer to an Engine.
+func New(title string) *Engine {
+	ctx := Engine{
```
Comment (aerwin3) [inline engine.go]: why not name this engine instead of ctx. That way the docs match the definition.

**Comment by aerwin3** [comment]
LGTM other than the tiny nit pick. :)

**Comment by hurricanerix** [comment]
We could do that, but I would prefer 'e' or 'eng' to 'engine'.

**Comment by aerwin3** [comment]
eng works great.

**Comment by hurricanerix** [comment]
@aerwin3 Changed ctx to e

**Pull request merged**